aten/src/ATen/SparseTensorImpl.h

```diff
@@ -183,13 +183,18 @@ struct CAFFE2_API SparseTensorImpl : public TensorImpl {
   // make it happen
   void set_indices_and_values_unsafe(const Tensor& indices, const Tensor& values);
 
-  // NOTE: `shallow_copy_and_detach()` does not copy the AutogradMeta pointer
-  // because it is unique for each Variable.
+  // NOTE: `shallow_copy_and_detach()` does not copy the following TensorImpl fields:
+  // 1. the AutogradMeta pointer, because it is unique for each Variable.
+  // 2. the version counter, because although it lives in TensorImpl, the version counter is managed
+  // by autograd, and the call sites of `shallow_copy_and_detach()` (from autograd) should decide what
+  // the version counter should be for each new TensorImpl. See NOTE [ Version Counter Sharing ] for details.
```

Comment:
could you add this comment for OpaqueTensorImpl as well?

Reply:
I'm not a big fan of copy pasting... it would be better to have a `See Note [blah blah]` cross-reference instead